### ISSUE TITLE
Given one input tensor, why tf.nn.max_pool generate two tensors?

### ISSUE THREAD

**Issue by leihe001** [tensorflow/tensorflow#3402]
with tf.name_scope('layer3'):
    conv3_weights = tf.Variable(tf.truncated_normal(
      [3, 3, 128, 256], stddev=0.1))
    variable_summaries(conv3_weights, 'layer3' + '/weights')
    conv3_biases = tf.Variable(tf.constant(0.1, shape=[256]))
    variable_summaries(conv3_biases, 'layer3' + '/biases')
    conv3 = tf.nn.conv2d(pool2,
      conv3_weights,
      strides=[1, 1, 1, 1],
      padding='SAME')
    tf.histogram_summary('layer3'+'/pre_activations', conv3)
    relu3 = tf.nn.relu(tf.nn.bias_add(conv3, conv3_biases))
    tf.histogram_summary('layer3'+'/activations', relu3)
    print relu3.get_shape()
    pool3 = tf.nn.max_pool(relu3,
      ksize=[1, 2, 2, 1],
      strides=[1, 2, 2, 1],
      padding='SAME')
    tf.histogram_summary('layer3'+'/pool', pool3)
    print pool3.get_shape()

When using TensorBoard to visualize the graph, I find the output of the layer3 are two tensors. I am puzzled why does it happen? Thank you very much! 
![1](https://cloud.githubusercontent.com/assets/20062851/16974093/b96aee78-4e6b-11e6-97c5-c266b4dc28d8.jpg)

**Comment by concretevitamin** [comment]
Can you click on `MaxPool` and see the `Outputs`?

I suspect the extra tensor might just be CheckNumerics or a control dependency?

**Comment by leihe001** [comment]
OK, the screenshot is as follow:
![2](https://cloud.githubusercontent.com/assets/20062851/16979506/fb42e134-4e93-11e6-9b40-3062fc421f27.jpg)

**Comment by leihe001** [comment]
Got it! I implement unpooling function in the layer4.

```
def UnPooling2x2ZeroFilled(x):
  out = tf.concat(3, [x, tf.zeros_like(x)])
  out = tf.concat(2, [out, tf.zeros_like(out)])

  sh = x.get_shape().as_list()
  if None not in sh[1:]:
  	out_size = [-1, sh[1] * 2, sh[2] * 2, sh[3]]
  	return tf.reshape(out, out_size)
  else:
	sh = tf.shape(x)
	ret = tf.reshape(out, tf.pack([-1, sh[1] * 2, sh[2] * 2, sh[3]]))
	ret.set_shape([None, None, None, sh[3]])
	return ret
```def UnPooling2x2ZeroFilled(x):
  out = tf.concat(3, [x, tf.zeros_like(x)])
  out = tf.concat(2, [out, tf.zeros_like(out)])

  sh = x.get_shape().as_list()
  if None not in sh[1:]:
  	out_size = [-1, sh[1] * 2, sh[2] * 2, sh[3]]
  	return tf.reshape(out, out_size)
  else:
	sh = tf.shape(x)
	ret = tf.reshape(out, tf.pack([-1, sh[1] * 2, sh[2] * 2, sh[3]]))
	ret.set_shape([None, None, None, sh[3]])
	return ret

where x and tf.zeros_like(x) are two tensors, is right? In addition, when the gradient is computed , the error raises as follow:

![3](https://cloud.githubusercontent.com/assets/20062851/16981952/d60e861e-4e9f-11e6-9400-e830989b885c.jpg)

Could you tell me what's wrong? Thank you very much!

**Comment by leihe001** [comment]
![3](https://cloud.githubusercontent.com/assets/20062851/16981976/ecfd6d36-4e9f-11e6-904c-05e804864f67.jpg)

**Comment by concretevitamin** [comment]
This does not sound like a bug ticket.  Please redirect to StackOverflow or tensorflow-discuss@.
